check_zpool.sh

```diff
@@ -4,9 +4,64 @@
 # Get your copy from https://github.com/Klintrup/check_zpool
 
 PATH="/sbin:/bin:/usr/sbin:/usr/bin"
+ERR="0"
 unset ERRORSTRING
 unset OKSTRING
-unset ERR
+
+validate_email() {
+  if [ -n "$_validate_email__input" ]; then
+    echo "_validate_email__input variable is already set" >&2
+    exit 1
+  fi
+  for _validate_email__input in "$@"; do
+    if ! echo "$_validate_email__input" | grep -qE '^[a-zA-Z0-9._%+-]{1,}@[a-zA-Z0-9.-]{1,}\.[a-zA-Z]{2,}$'; then
+      echo "$_validate_email__input is not a valid email address" >&2
+    else
+      echo "$_validate_email__input"
+    fi
+  done
+}
```

Comment:
Local variables are not supported in POSIX sh, and thus throws a linter error when used.

Reply:
That is exactly what I've done, except I have used an even more obscure variable name, and included the function name in the variable name to avoid additional issues.

How is using email_addr better than _validate_email__input ?

I also just added code to unset the variable at the end of the function to be 100% sure it doesnt leak anything.